src/main/java/gmail/mishchuk/modultest_2/PhoneBook.java

```diff
@@ -0,0 +1,64 @@
+package gmail.mishchuk.modultest_2;
+
+import lombok.Data;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Data
+public class PhoneBook {
+
+    private Map<String, String> contacts = new HashMap<>();
+
+    //можна було б і не створювати пустий конструктор
+    public PhoneBook() {
+    }
+
+    public void addContact(String name, String phone) {
+        contacts.put(name, phone);
+        System.out.println("Створено контакт:\n\t"
+                + name + ", номер: "
+                + phone);
+    }
+
+    public void updateContact(String name, String newPhone) {
+        if (contacts.containsKey(name)) {
+            contacts.put(name, newPhone);
+            System.out.println("\nКонтакт оновлено:\n\t"
+                    + name + ", новий номер: "
+                    + newPhone);
+        } else {
+            addContact(name, newPhone);
+        }
+    }
+
+    public void deleteContact(String name) {
+        if (contacts.containsKey(name)) {
+            String removedContact = contacts.remove(name);
+            System.out.println("\nУспішно видалено!\n\tІм'я: "
+                    + name + ", номер: "
+                    + removedContact);
+        }
+    }
```

Comment:
Можна просто записати без перевірки на наявність адже .remove не викидає помилку якщо елемент відсутній.

```suggestion
    public void deleteContact(String name) {
        String removedContact = contacts.remove(name);
        if (removedContact != null) {
            System.out.println("\nУспішно видалено!\n\tІм'я: "
                    + name + ", номер: "
                    + removedContact);
        } else {
            System.out.println("\nУспішно видалено!\n\tІм'я: " + name);
        }
    }
```